src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -950,7 +950,7 @@ public abstract sealed class $Type$Buffer
         return get(index, dst, 0, dst.length);
     }
 
-    private $Type$Buffer getArray(int index, $type$[] dst, int offset, int length) {
+    $Type$Buffer getArray(int index, $type$[] dst, int offset, int length) {
         if (
 #if[char]
             isAddressable() &&
```

Comment:
is `isAddressable` only relevant for `StringCharBuffer`?
If so, this check becomes effectively dead code with `StringCharBuffer` overriding the method.
Perhaps an assert statement instead to avoid any future modification breaking the assumption?

Reply:
> Perhaps an assert statement instead to avoid any future modification breaking the assumption?

I added an `assert` in 3c88dac but I am not entirely sure it's the one you had in mind.

Reply:
>  I am not entirely sure it's the one you had in mind.

:) yeah - I was thinking more like this:

```
    $Type$Buffer getArray(int index, $type$[] dst, int offset, int length) {
#if[char]
        assert !isAddressable();
#end[char]
        if (((long)length << $LG_BYTES_PER_VALUE$) > Bits.JNI_COPY_TO_ARRAY_THRESHOLD) {
```

But it was really just an idea to remove a check which should always be true now that overridden.

Reply:
> #if[char]
>         assert !isAddressable();
> #end[char]

I think you intended `isAddressable()` in the above, not its negation.